compiler/internal/analysis/info.go

```diff
@@ -248,16 +228,27 @@ type FuncInfo struct {
 	// List of continue statements in the function.
 	continueStmts []continueStmt
 	// List of return statements in the function.
-	returnStmts []astPath
+	returnStmts []returnStmt
+	// List of most of the deferred function calls in the function.
```

Comment:
Why "most"?  Which ones are excluded and why?

Reply:
We should be able record every `defer` but I didn't because the `defer`s that aren't blocking aren't needed when determining if a `return` is blocking. So, since calls from `js.Object`s and builtin functions (e.g. `len`) aren't blocking, I didn't add those. Instead I added some comments on [line 472](https://github.com/gopherjs/gopherjs/pull/1349/files#diff-8ca33eb0bc7ed01cd267a73c3e2557daa5b9bb71b5ebb20b81fb1fb7b691e668R472) and [line 623](https://github.com/gopherjs/gopherjs/pull/1349/files#diff-8ca33eb0bc7ed01cd267a73c3e2557daa5b9bb71b5ebb20b81fb1fb7b691e668R623) in info.go where we possibly are skipping defers. We could add them with `if forDefer { fi.deferStmts = append(fi.deferStmts, newDefer(false)) }` in those cases.

Since we still needed to know if there were any defers, I left the `HasDefer` as it was.

Reply:
Maybe rephrase it as "List of deferred function calls which could be blocking"?

Reply:
Changed